src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3270,6 +3277,42 @@ internal bool CanEmitSpecialType(SpecialType type)
             return (diagnostic == null) || (diagnostic.Severity != DiagnosticSeverity.Error);
         }
 
+        internal bool EmitNullablePublicOnly
+        {
+            get
+            {
+                if (!_lazyEmitNullablePublicOnly.HasValue())
+                {
+                    bool value = SyntaxTrees.FirstOrDefault()?.Options?.Features?.ContainsKey("nullablePublicOnly") == true;
```

Comment:
>SyntaxTrees.FirstOrDefault()?.Options?.Features?.ContainsKey("nullablePublicOnly") == true; [](start = 33, length = 91)

We may already have a way of checking feature flags. I'm surprised we'd need to do this "by hand" here.

Reply:
I couldn't find an appropriate helper method for checking `Features`. 

Regarding language version, is it necessary to add that restriction here? Presumably language version will produce the necessary warnings and errors during binding.

---
In reply to: [294014629](https://github.com/dotnet/roslyn/pull/36398#discussion_r294014629) [](ancestors = 294014629)

Reply:
You're right, we don't have a helper. IOperations did the same...

```C#
        internal override bool IsIOperationFeatureEnabled()
        {
            var options = (CSharpParseOptions)this.SyntaxTrees.FirstOrDefault()?.Options;
            return options?.IsFeatureEnabled(MessageID.IDS_FeatureIOperation) ?? false;
        }
```

For LangVersion, I agree too. If you're in C# 7.3 and don't nullable annotations, then the feature flag is a no-op, so no need to report a diagnostic.

---
In reply to: [294017389](https://github.com/dotnet/roslyn/pull/36398#discussion_r294017389) [](ancestors = 294017389,294014629)